Assets/Tests/InputSystem/Plugins/InputForUITests.cs

```diff
@@ -127,78 +110,389 @@ public void TouchIsPressedAndMouseEventsAreIgnored()
+    [TestCase(true)]
+    [TestCase(false)]
+    public void UI_Input_CancelActionWorks(bool useProjectWideActionsAsset)
     {
+        Update();
+        if (!useProjectWideActionsAsset)
+        {
+            InputSystem.s_Manager.actions = null;
+        }
+        Update();
+
         var gamepad = InputSystem.AddDevice<Gamepad>();
+        var keyboard = InputSystem.AddDevice<Keyboard>();
+        PressUpdateReleaseUpdate(gamepad.buttonEast);
+        PressUpdateReleaseUpdate(keyboard.escapeKey);
+
+        Assert.AreEqual(2, m_InputForUIEvents.Count);
+
+        Assert.That(m_InputForUIEvents[0] is
+        {
+            type: Event.Type.NavigationEvent,
+            asNavigationEvent: { type: NavigationEvent.Type.Cancel, eventSource: EventSource.Gamepad }
+        });
+        Assert.That(m_InputForUIEvents[1] is
+        {
+            type: Event.Type.NavigationEvent,
+            asNavigationEvent: { type: NavigationEvent.Type.Cancel, eventSource: EventSource.Keyboard }
+        });
+    }
+
+    [Test]
+    [Category(kTestCategory)]
+    [TestCase(true)]
+    [TestCase(false)]
+    public void UI_Input_PointActionWorks(bool useProjectWideActionsAsset)
+    {
         Update();
-        Press(gamepad.leftStick.left);
+        if (!useProjectWideActionsAsset)
+        {
+            InputSystem.s_Manager.actions = null;
+        }
         Update();
-        Release(gamepad.leftStick.left);
+
+        var mouse = InputSystem.AddDevice<Mouse>();
+        Set(mouse.position, new Vector2(0.5f, 0.5f));
         Update();
+        Move(mouse.position, new Vector2(100f, 0.5f));
+        Update();
+
+        var pen = InputSystem.AddDevice<Pen>();
+        Set(pen.position, new Vector2(0.5f, 0.5f));
+        Update();
+        Move(pen.position, new Vector2(100f, 0.5f));
+        Update();
+
+        var touch = InputSystem.AddDevice<Touchscreen>();
+        BeginTouch(1, new Vector2(0.5f, 0.5f));
+        Update();
+        EndTouch(1, new Vector2(100f, 100f));
+        Update();
+
+        // Touch screen move is three actions: press, move, release
+        Assert.AreEqual(5, m_InputForUIEvents.Count);
+        Assert.That(m_InputForUIEvents[0] is
+        {
+            type: Event.Type.PointerEvent,
+            asPointerEvent: { type: PointerEvent.Type.PointerMoved, eventSource: EventSource.Mouse }
+        });
+        Assert.That(m_InputForUIEvents[1] is
+        {
+            type: Event.Type.PointerEvent,
+            asPointerEvent: { type: PointerEvent.Type.PointerMoved, eventSource: EventSource.Pen }
+        });
+        Assert.That(m_InputForUIEvents[3] is
+        {
+            type: Event.Type.PointerEvent,
+            asPointerEvent: { type: PointerEvent.Type.PointerMoved, eventSource: EventSource.Touch }
+        });
     }
 
     [Test]
     [Category(kTestCategory)]
-    public void SendWheelEvent()
+    [TestCase(true)]
+    [TestCase(false)]
+    public void UI_Input_ClickActionWorks(bool useProjectWideActionsAsset)
```

Comment:
@graham-huws did you notice any difference in behavior on these clicks? I see left/middle/right is configured differently in the PWA asset

Reply:
Not sure what you mean here sorry - They produce the different PointerEvent.Button (left/middle/right), is there something else I should be checking for?

Reply:
Sorry for vagueness, I was basically referring to the fact that they have different configuration...

Click has initialStateCheck = true while Middle and Right Clicks have initialStateCheck = false, doesn't make sense to me....